File: leet_easy_c++/leet_111/leet_111a.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    int minDepth(TreeNode* root) {
        queue<TreeNode*> q;
        q.push(root);
        if(!root) return 0; if(root->left == NULL && root->right == NULL) return 1;
        int next, cnt = 0, level=1, flag = 1;
        while(flag == 1){
            next=0;
            for(int i=0; flag == 1 && i< level; i++){
                if(q.front() != NULL && q.front()->left != NULL){
                    q.push(q.front()->left);
                    next++;
                }
                if(q.front() != NULL && q.front()->right != NULL){
                    q.push(q.front()->right);
                    next++;
                }
                if(q.front() != NULL && q.front()->right == NULL && q.front()->left == NULL){
                    flag = 0;
                }
                q.pop();
            }
            level = next;
            cnt++;
        }
        return cnt;
    }
};

File: leet_easy_python/leet_111/leet_111a.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def minDepth(self, root):
        if not root: return 0
        if root.left and root.right: return min(self.minDepth(root.left), self.minDepth(root.right)) + 1
        elif root.left: return self.minDepth(root.left) + 1
        else: return self.minDepth(root.right) + 1
        
        """
        :type root: TreeNode
        :rtype: int
        """
        